packages/safe-tools-client/app/components/setup-safe-modal/index.gts

```diff
@@ -68,30 +61,36 @@ export default class SetupSafeModal extends Component<Signature> {
                fee.
             </li>
           </ul>
-          <b>Estimated gas cost: {{this.gasCost}}</b>
-          <div class='safe-setup-modal__section-wallet-info'>
-            {{svgJar
-              (if this.notEnoughBalance 'icon-x-circle-ht' 'icon-check-circle-ht')
-              class='safe-setup-modal__section-icon'
-              style=(cssVar
-                icon-color=(if this.notEnoughBalance 'var(--boxel-red)' 'var(--boxel-green)')
-              )
-            }}
-            <p>
-              Your wallet has {{if this.notEnoughBalance 'in'}}sufficient funds to cover the estimated gas cost.
-            </p>
-          </div>
+          {{#if @isLoadingGasInfo}} 
+            <div class='safe-setup-modal__section__section-gas-loading'>
+              <BoxelLoadingIndicator /> <p>Calculating estimated gas cost.</p>
+            </div>
+          {{else}}
+            <b>Estimated gas cost: {{@gasCostDisplay}}</b>
+            <div class='safe-setup-modal__section-wallet-info'>
+              {{svgJar
+                (if this.notEnoughBalance 'icon-x-circle-ht' 'icon-check-circle-ht')
+                class='safe-setup-modal__section-icon'
+                style=(cssVar
+                  icon-color=(if this.notEnoughBalance 'var(--boxel-red)' 'var(--boxel-green)')
+                )
+              }}
+              <p>
+                Your wallet has {{if this.notEnoughBalance 'in'}}sufficient funds to cover the estimated gas cost.
+              </p>
+            </div>
+          {{/if}}
         </Section>
         <ActionChin @state='default'>
           <:default as |ac|>
             <ac.ActionButton
-              @loading={{this.provisioning}}
-              disabled={{this.notEnoughBalance}}
-              {{on 'click' this.createSafe}}
+              @loading={{@provisioning}}
+              disabled={{or this.notEnoughBalance @isLoadingGasInfo}}
+              {{on 'click' @onProvisionClick}}
             >
-              Provision{{if this.provisioning 'ing'}}
+              Provision{{if @provisioning 'ing'}}
```

Comment:
ActionChin has an in-progress state that would be suitable to use while the provisioning is occurring.

Reply:
wil be applying that on my next pr which will handle the safe provisioning 🙂